dhall/src/Dhall.hs

```diff
@@ -1092,17 +1155,13 @@ pair :: Decoder a -> Decoder b -> Decoder (a, b)
 pair l r = Decoder extractOut expectedOut
   where
     extractOut expr@(RecordLit fields) =
-      (,) <$> ( Data.Maybe.maybe (typeError expectedOut expr) (extract l) $ Dhall.Map.lookup "_1" fields)
-          <*> ( Data.Maybe.maybe (typeError expectedOut expr) (extract r) $ Dhall.Map.lookup "_2" fields)
-    extractOut expr = typeError expectedOut expr
-
-    expectedOut =
-        Record
-            (Dhall.Map.fromList
-                [ ("_1", expected l)
-                , ("_2", expected r)
-                ]
-            )
+      (,) <$> Data.Maybe.maybe (typeError' expectedOut expr) (extract l) (Dhall.Map.lookup "_1" fields)
+          <*> Data.Maybe.maybe (typeError' expectedOut expr) (extract r) (Dhall.Map.lookup "_2" fields)
+    extractOut expr = typeError' expectedOut expr
+
+    expectedOut = (\l' r' -> Record (Dhall.Map.fromList [("_1", l'), ("_2", r')]))
```

Comment:
This can also be simplified using `ApplicativeDo`

Reply:
Done.